File: Hello/HelloWidget.cpp
#include <QLabel>
#include <QLineEdit>
#include <QTextEdit>
#include <QPushButton>
#include <QHBoxLayout>
#include <QVBoxLayout>

#include "HelloWidget.h"

HelloWidget::HelloWidget(QWidget * parent) :
    QWidget(parent)
{
    QVBoxLayout * mainLayout = new QVBoxLayout(this);
    QHBoxLayout * nameLayout = new QHBoxLayout();
    
    nameLabel = new QLabel("Name: ");
    nameLayout->addWidget(nameLabel);
    
    nameLineEdit = new QLineEdit();
    nameLayout->addWidget(nameLineEdit);
    mainLayout->addLayout(nameLayout);
    
    logTextEdit = new QTextEdit();
    logTextEdit->setReadOnly(true);
    mainLayout->addWidget(logTextEdit);
    
    helloButton = new QPushButton("Say hello!");
    mainLayout->addWidget(helloButton);
    
    connect(helloButton, SIGNAL(clicked()), this, SLOT(sayHello()));
}

HelloWidget::~HelloWidget()
{
}

///////////////////
// Private slots //
///////////////////

void HelloWidget::sayHello()
{
    if (!nameLineEdit->text().isEmpty()) {
        logTextEdit->append("Hello " + nameLineEdit->text());
    }
}

File: Hello/python/HelloWidget.py
# encoding: utf-8

from PyQt4 import QtGui

class HelloWidget(QtGui.QWidget):

    def __init__(self, parent = None):
        super(QtGui.QWidget, self).__init__(parent)
        mainLayout = QtGui.QVBoxLayout(self)
        nameLayout = QtGui.QHBoxLayout()
        
        self.nameLabel = QtGui.QLabel('Name:')
        self.nameLineEdit = QtGui.QLineEdit()
        nameLayout.addWidget(self.nameLabel)
        nameLayout.addWidget(self.nameLineEdit)
        mainLayout.addLayout(nameLayout)
        
        self.logTextEdit = QtGui.QTextEdit()
        self.logTextEdit.setReadOnly(True)
        mainLayout.addWidget(self.logTextEdit)
        
        self.helloButton = QtGui.QPushButton('Say hello:')
        mainLayout.addWidget(self.helloButton)
        
        self.helloButton.clicked.connect(self.sayHello)
        
    def sayHello(self):
        if self.nameLineEdit.text():
            self.logTextEdit.append("Hello " + self.nameLineEdit.text() + "!")

